platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1025,35 +700,53 @@ namespace platoon_strategic
         }
     }
 
-    void PlatoonStrategicPlugin::mob_op_cb_leader(const cav_msgs::MobilityOperation& msg)
+    // UCLA: add condition to account for frontal joining 
+    void PlatoonStrategicIHPPlugin::mob_op_cb_leader(const cav_msgs::MobilityOperation& msg)
     {   
         std::string strategyParams = msg.strategy_params;
-        std::string senderId = msg.header.sender_id;
-        std::string platoonId = msg.header.plan_id;
+        std::string senderId = msg.header.sender_id; //sender is target platoon leader
+        std::string platoonId = msg.header.plan_id; // platoon ID is the target platoon 
+        // frontal join
+        std::string newLeaderID = msg.header.sender_id; // new leader is the vehicle in front 
+        std::string newLeaderPlatoonId = pm_.currentPlatoonID; // set self's (front join veh) platoon ID as new platoon ID
         // In the current state, we care about the INFO heart-beat operation message if we are not currently in
         // a negotiation, and also we need to care about operation from members in our current platoon
 
         bool isPlatoonInfoMsg = (strategyParams.rfind(OPERATION_INFO_TYPE, 0) == 0);
         bool isPlatoonStatusMsg = (strategyParams.rfind(OPERATION_STATUS_TYPE, 0) == 0);
         bool isNotInNegotiation = (pm_.current_plan.valid == false);
+
+        // INFO params
```

Comment:
This comment really needs to explain what decision is being made here (in English).  The first condition indicates the message is coming from a leader or candidate leader, and the second is unnecessarily confusing double-negative logic that says host is currently in a platoon.  Therefore, this is the decision whether I am the leader hearing from a joining candidate or whether I am the joining candidate hearing from the current leader!  Nothing above this line can assume which of those is the case.  A suggested readability improvement would be:

bool isPlatoonInfoMsg = (strategyParams.rfind(OPERATION_INFO_TYPE, 0) == 0); //only comes from leader or candidate leader
bool iAmLeader = pm_.current_plan.valid

//if I am solo vehicle (candidate leader) hearing from a platoon leader then
if (isPlatoonInfoMsg  &&  !iAmLeader){
...